Update documentation and move pre-requisites to the main README.md (#15)
Azure-Samples/java-ai: README.md

![Functional Architecture](./media/functional-architecture.png)

## Prerequisites

> Easiest way to start is to [deploy a GitHub Codespace](https://github.com/codespaces/new/Azure-Samples/java-ai) as it contains all the prerequisites.

Reviewer: I think they'll have to Fork it first.

```suggestion
Easiest way to start is to Fork and [open the repository in a GitHub Codespace](https://github.com/codespaces/new/Azure-Samples/java-ai) as it contains all the prerequisites.
```

> If you deploy locally, you need to have the following installed:

Reviewer: ```suggestion
If you develop locally, you need to have the following installed:
```

> - [Java 17+](https://learn.microsoft.com/en-us/java/openjdk/download)

Reviewer: ```suggestion
- [Java 17+](https://learn.microsoft.com/java/openjdk/download)
```

> - [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli)

Reviewer: ```suggestion
- [Azure CLI](https://docs.microsoft.com/cli/azure/install-azure-cli)
```

The following resources will be created as represented in the diagram below:
- Resource Group
- Azure Container Registry

> - [Azure Container Apps Environment](https://learn.microsoft.com/en-us/azure/container-apps/)

Reviewer: ```suggestion
- [Azure Container Apps Environment](https://learn.microsoft.com/azure/container-apps/)
```

>     - [Eureka Server for Spring Component](https://learn.microsoft.com/en-us/azure/container-apps/java-eureka-server-usage)

Reviewer: ```suggestion
    - [Eureka Server for Spring Component](https://learn.microsoft.com/azure/container-apps/java-eureka-server-usage)
```

>     - [Admin for Spring Component](https://learn.microsoft.com/en-us/azure/container-apps/java-admin-for-spring-usage)

Reviewer: ```suggestion
    - [Admin for Spring Component](https://learn.microsoft.com/azure/container-apps/java-admin-for-spring-usage)
```

- Azure Blob Storage Account
- Azure Key Vault for the `item-category-service` Azure OpenAI API Key secret
- User-assigned Managed Identity to pull the images from the Azure Container Registry
- Azure OpenAI

> - [Azure Container Apps with Java Runtime](https://learn.microsoft.com/en-us/azure/container-apps/java-metrics)

Reviewer: ```suggestion
- [Azure Container Apps with Java Runtime](https://learn.microsoft.com/azure/container-apps/java-metrics)
```
